docs/src/main/tut/docs/core/algebras/README.md

```diff
@@ -97,41 +82,102 @@ object UserRepository {
 }
 ```
 
-Let's examine the two fragments above to understand what Freestyle is doing for you.
+Let us examine how the `@free`-annotated trait relates to the code generated from it. 
+
+### Generalise Trait
+
+The `@free` macro makes two changes to the trait `Users`.
 
-## Automatic method implementations
+1. It adds a type parameter `F[_]`, of kind `* -> *`, which is a general type in which operations are constructed. This makes `Users[F[_]]` resemble a [_Generalised Abstract Data Type (GADT)_](https://pchiusano.github.io/2014-05-20/scala-gadts.html), an interface to construct the algebra's operations into a type `F[_]`.
+
+2. It makes `Users` to extend `EffectLike[F[_]]`. This trait defines the type `FS[_]`, used in the smart constructors, as `FS[_] = FreeS.Par[F]`, which in turn is an alias for `Free[FreeApplicative[F, ?], ?]`.
```

Comment:
`FreeS.Par[F, A]` is a type alias for `FreeApplicative[F, A]`,
`FreeS[F, A]`(`FS.Seq`) is `Free[FreeApplicative[F, ?], A]`.

Like you mentioned below.